From 2944386f64853fbc508e870223086c2f1457150c Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 14 Jul 2014 13:41:43 -0400
Subject: [x86] hwpoison: remove unused global variable in do_machine_check()

Message-id: <1405345303-24468-14-git-send-email-prarit@redhat.com>
Patchwork-id: 85777
O-Subject: [RHEL7.1 PATCH BZ 1119306 13/13] hwpoison: remove unused global variable in do_machine_check()
Bugzilla: 1119306
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119306

commit 65eb71823b01051ca6e256e9cc8259141a849052
Author: Chen Yucong <slaoub@gmail.com>
Date:   Wed Jun 4 16:10:43 2014 -0700

    hwpoison: remove unused global variable in do_machine_check()

    Remove an unused global variable mce_entry and relative operations in
    do_machine_check().

    Signed-off-by: Chen Yucong <slaoub@gmail.com>
    Cc: Naoya Horiguchi <n-horiguchi@ah.jp.nec.com>
    Cc: Wu Fengguang <fengguang.wu@intel.com>
    Cc: Andi Kleen <andi@firstfloor.org>
    Cc: Ingo Molnar <mingo@elte.hu>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: "H. Peter Anvin" <hpa@zytor.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Cc: Dean Nelson <dnelson@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/mce.h b/arch/x86/include/asm/mce.h
index cbe6b9e..d077a7a 100644
--- a/arch/x86/include/asm/mce.h
+++ b/arch/x86/include/asm/mce.h
@@ -176,8 +176,6 @@ int mce_available(struct cpuinfo_x86 *c);
 DECLARE_PER_CPU(unsigned, mce_exception_count);
 DECLARE_PER_CPU(unsigned, mce_poll_count);
 
-extern atomic_t mce_entry;
-
 typedef DECLARE_BITMAP(mce_banks_t, MAX_NR_BANKS);
 DECLARE_PER_CPU(mce_banks_t, mce_poll_banks);
 
diff --git a/arch/x86/kernel/cpu/mcheck/mce.c b/arch/x86/kernel/cpu/mcheck/mce.c
index 6cc8003..bb92f38 100644
--- a/arch/x86/kernel/cpu/mcheck/mce.c
+++ b/arch/x86/kernel/cpu/mcheck/mce.c
@@ -60,8 +60,6 @@ static DEFINE_MUTEX(mce_chrdev_read_mutex);
 
 #define SPINUNIT 100 /* 100ns */
 
-atomic_t mce_entry;
-
 DEFINE_PER_CPU(unsigned, mce_exception_count);
 
 struct mce_bank *mce_banks __read_mostly;
@@ -1040,8 +1038,6 @@ void do_machine_check(struct pt_regs *regs, long error_code)
  DECLARE_BITMAP(valid_banks, MAX_NR_BANKS);
  char *msg = "Unknown";
 
- atomic_inc(&mce_entry);
-
  this_cpu_inc(mce_exception_count);
 
  if (!cfg->banks)
@@ -1171,7 +1167,6 @@ void do_machine_check(struct pt_regs *regs, long error_code)
   mce_report_event(regs);
  mce_wrmsrl(MSR_IA32_MCG_STATUS, 0);
 out:
- atomic_dec(&mce_entry);
  sync_core();
 }
 EXPORT_SYMBOL_GPL(do_machine_check);
-- 
1.7.1